web/src/components/data-fields/address/AddressMunicipalityDropdown.tsx

```diff
@@ -0,0 +1,46 @@
+import { MunicipalityDropdown } from "@/components/data-fields/MunicipalityDropdown";
+import { AddressContext } from "@/contexts/addressContext";
+import { MunicipalitiesContext } from "@/contexts/municipalitiesContext";
+import { ProfileFormContext } from "@/contexts/profileFormContext";
+import { useAddressErrors } from "@/lib/data-hooks/useAddressErrors";
+import { useFormContextFieldHelpers } from "@/lib/data-hooks/useFormContextFieldHelpers";
+import { Address, Municipality } from "@businessnjgovnavigator/shared/";
+import { ReactElement, useContext } from "react";
+
+export const AddressMunicipalityDropdown = (): ReactElement => {
+  const { state, setAddressData } = useContext(AddressContext);
+  const { municipalities } = useContext(MunicipalitiesContext);
+  const { doesFieldHaveError, getFieldErrorLabel } = useAddressErrors();
+
+  const { setIsValid: setIsValidAddressLine1 } = useFormContextFieldHelpers(
+    "addressLine1",
+    ProfileFormContext
+  );
+  const { setIsValid: setIsValidMunicipality } = useFormContextFieldHelpers(
+    "addressMunicipality",
+    ProfileFormContext
+  );
+  const { setIsValid: setIsValidZipCode } = useFormContextFieldHelpers("addressZipCode", ProfileFormContext);
+
+  const onSelect = (value: Municipality | undefined): void => {
+    setAddressData((previousAddressData: Address): Address => {
+      return { ...previousAddressData, addressMunicipality: value };
+    });
+  };
+
+  return (
+    <MunicipalityDropdown
+      onValidation={(): void => {
+        setIsValidAddressLine1(!doesFieldHaveError("addressLine1"));
+        setIsValidMunicipality(!doesFieldHaveError("addressMunicipality"));
+        setIsValidZipCode(!doesFieldHaveError("addressZipCode"));
+      }}
```

Comment:
Noted in `profile-owning.test.tsx` - add coverage in the case user provides Address line 2 and no other data. Partial address error messages should display.